### PR TITLE
Add user to team API

### PR DESCRIPTION
**Pull request by yijia-cc**

### CODE REVIEW

File: app/api/gql/resolver/invitation.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import (
+	"github.com/graph-gophers/graphql-go"
+	"github.com/teamyapp/teamy-backend/app/entity"
+)
+
+type Invitation struct {
+	deps          *Dependencies
+	Entity
+	invitation entity.Invitation
+	query      Query
```
Comment (naturetime10) [inline app/api/gql/resolver/invitation.go]:
```suggestion
	deps       *Dependencies
	query      Query
	Entity
	invitation entity.Invitation
```

File: app/api/gql/resolver/invitation.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+func (i Invitation) NewMember() *User {
+	panic("need implementation")
+}
+
+func (i Invitation) NewMemberEmail() *string {
+	panic("need implementation")
+}
+
+func (i Invitation) Team() Team {
```
Comment (naturetime10) [inline app/api/gql/resolver/invitation.go]:
```suggestion
func (i Invitation) TeamToJoin() Team {
```

File: app/api/gql/resolver/invitation.go [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+	panic("need implementation")
+}
+
+func (i Invitation) Expiration() graphql.Time {
```
Comment (naturetime10) [inline app/api/gql/resolver/invitation.go]:
```suggestion
func (i Invitation) ExpireAt() graphql.Time {
```

File: app/api/gql/resolver/invitation_status.go [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package resolver
+
+type InvitationStatus string
+
+const (
+	Pending  InvitationStatus = "PENDING"
+	Accepted InvitationStatus = "ACCEPTED"
+	Declined InvitationStatus = "DECLINED"
+	Revoked  InvitationStatus = "REVOKED"
+	Expired  InvitationStatus = "EXPIRED"
```
Comment (naturetime10) [inline app/api/gql/resolver/invitation_status.go]:
```suggestion
	invitationStatusPending  InvitationStatus = "PENDING"
	invitationStatusAccepted InvitationStatus = "ACCEPTED"
	invitationStatusDeclined InvitationStatus = "DECLINED"
	invitationStatusRevoked  InvitationStatus = "REVOKED"
	invitationStatusExpired  InvitationStatus = "EXPIRED"
```

File: app/api/gql/resolver/mutation.go [hunk 5]
```diff
@@ -421,6 +421,42 @@ func contains(arr []oneEntity.ID, element oneEntity.ID) bool {
...
 
+func (m Mutation) AddUserToTeam(ctx context.Context, args struct {
+	UserID graphql.ID
+	TeamID graphql.ID
+}) (bool, error) {
+	panic("not implemented")
+}
+
+func (m Mutation) CreateInvitation(ctx context.Context, args struct {
+	ReceiveInviteUserEmail string
```
Comment (naturetime10) [inline app/api/gql/resolver/mutation.go]:
```suggestion
	InvitedUserEmail string
```

File: app/api/gql/schema.graphqls [hunk 6]
```diff
@@ -144,6 +150,25 @@ enum TaskStatus {
 	DELIVERED
 }
 
+type Invitation implements Entity {
+	id: ID!
+	inviter: User!
+	newMember: User
+	newMemberEmail: String
+	team: Team!
+	expiration: Time!
```
Comment (naturetime10) [inline app/api/gql/schema.graphqls]:
```suggestion
	expireAt: Time!
```

File: app/entity/invitation.go [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+import (
+	"time"
+
+	oneEntity "github.com/teamyapp/one/entity"
+)
+
+type Invitation struct {
+	oneEntity.Entity
+	SendInviteUserID       oneEntity.ID
```
Comment (naturetime10) [inline app/entity/invitation.go]:
```suggestion
	InviterUserID       oneEntity.ID
```

File: app/entity/invitation.go [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+	ReceiveInviteUserID    oneEntity.ID
```
Comment (naturetime10) [inline app/entity/invitation.go]:
```suggestion
	NewMemberUserID    oneEntity.ID
```

File: app/entity/invitation.go [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+	ReceiveInviteUserEmail string
```
Comment (naturetime10) [inline app/entity/invitation.go]:
```suggestion
	NewMemberEmail string
```

File: app/entity/invitation.go [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+	TeamID                 oneEntity.ID
+	Expiration             time.Time
```
Comment (naturetime10) [inline app/entity/invitation.go]:
```suggestion
	TTL             time.Duration
```

File: app/entity/invitation_status.go [hunk 11]
```diff
@@ -0,0 +1,11 @@
+package entity
+
+type InvitationStatus int
+
+const (
+	Pending InvitationStatus = iota
+	Accepted
+	Declined
+	Revoked
+	Expired
```
Comment (naturetime10) [inline app/entity/invitation_status.go]:
```suggestion
	InvitationStatusPending InvitationStatus = iota
	InvitationStatusAccepted 
	InvitationStatusDeclined
	InvitationStatusRevoked
	InvitationStatusExpired
```

File: app/api/gql/resolver/mutation.go [hunk 12]
```diff
@@ -421,6 +421,42 @@ func contains(arr []oneEntity.ID, element oneEntity.ID) bool {
 	return false
 }
 
+func (m Mutation) AddUserToTeam(ctx context.Context, args struct {
+	UserID graphql.ID
+	TeamID graphql.ID
```
Comment (naturetime10) [inline app/api/gql/resolver/mutation.go]:
```suggestion
	UserID graphql.ID
	TeamID graphql.ID
	AcceptBeforeJoin bool
```

**Review by naturetime10 — CHANGES_REQUESTED** [review]

File: app/api/gql/resolver/invitation.go (hunk 1 above)
Comment (yijia-cc) [reply app/api/gql/resolver/invitation.go]: can I put Entity at the top?

File: app/entity/invitation.go [hunk 13]
```diff
@@ -0,0 +1,17 @@
...
+
+import (
+	"time"
+
+	oneEntity "github.com/teamyapp/one/entity"
+)
+
+type Invitation struct {
+	oneEntity.Entity
+	InviterUserID       oneEntity.ID
```
Comment (naturetime10) [inline app/entity/invitation.go]: Let's run `goimports -w .`

**Review by naturetime10 — CHANGES_REQUESTED** [review]

**Review by naturetime10 — APPROVED** [review]

**Pull request merged**